airflow/www/views.py

```diff
@@ -3772,3 +3772,102 @@ def autocomplete(self, session=None):
         payload = [row[0] for row in dag_ids_query.union(owners_query).limit(10).all()]
 
         return wwwutils.json_response(payload)
+
+
+class DagDependenciesView(AirflowBaseView):
+    """View to show dependencies between DAGs"""
+
+    refresh_interval = conf.getint(
+        "dag_dependencies_plugin",
+        "refresh_interval",
+        fallback=conf.getint("scheduler", "dag_dir_list_interval"),
+    )
+    last_refresh = datetime.utcnow() - timedelta(seconds=refresh_interval)
+    nodes = []
+    edges = []
+
+    @expose('/dag-dependencies')
+    @auth.has_access(
+        [
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+        ]
+    )
+    @gzipped
+    @action_logging
+    def list(self):
+        """Display DAG dependencies"""
+        title = "DAG Dependencies"
+
+        if datetime.utcnow() > self.last_refresh + timedelta(seconds=self.refresh_interval):
+            self._calculate_graph()
+            self.last_refresh = datetime.utcnow()
+
+        return self.render_template(
+            "airflow/dag_dependencies.html",
+            title=title,
+            nodes=self.nodes,
+            edges=self.edges,
+            last_refresh=self.last_refresh.strftime("%Y-%m-%d %H:%M:%S"),
+            arrange=conf.get("webserver", "dag_orientation"),
+            width=request.args.get("width", "100%"),
+            height=request.args.get("height", "800"),
+        )
+
+    def _calculate_graph(self):
+
+        current_app.dag_bag.collect_dags_from_db()
```

Comment:
I fear this call the most. Some DAGs can contain a thousand tasks, and we can have several such DAGs. It will be very slow to load them all into memory.  Do you think we can optimize it?

Reply:
Not quite sure how I could optimize it? It's already optimized in that it only refreshes after an interval which is equal to the scheduler recalculation interval. All dags are required to calculate the dependencies. Few ideas I have, which are not optimization but might help:

- set last_refresh_time at the start of the refresh, so that a second refresh triggered before first one finishes does not cause a recalculation in parallel
- remove dags without dependencies from rendering
- add a global configuration option to disable the view. I guess some deployments may be too large for such a global view.

Reply:
I am thinking of creating a new table that will contain only these relationships and try to be kept in good condition by DagFileProcessor.  This way, we will only be able to read the information we need.